GH-91079: Rename C_RECURSION_LIMIT to Py_C_RECURSION_LIMIT

vstinner opened the pull request:
Symbols of the C API should be prefixed by "Py_" to avoid conflict with existing names in 3rd party C extensions on #include <Python.h>.

<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
gh-NNNNN: Summary of the changes made
```

Where: gh-NNNNN refers to the GitHub issue number.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `main`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNN)
```

Where: [X.Y] is the branch name, e.g. [3.6].

GH-NNNN refers to the PR number from `main`.

-->


<!-- gh-issue-number: gh-91079 -->
* Issue: gh-91079
<!-- /gh-issue-number -->

vstinner committed:
GH-91079: Rename C_RECURSION_LIMIT to Py_C_RECURSION_LIMIT

Symbols of the C API should be prefixed by "Py_" to avoid conflict
with existing names in 3rd party C extensions on #include <Python.h>.

iritkatriel commented on Include/cpython/pystate.h:
```diff
@@ -197,12 +197,12 @@ struct _ts {
    layout, optimization, and WASI runtime. Wasmtime can handle about 700
    recursions, sometimes less. 500 is a more conservative limit. */
-#ifndef C_RECURSION_LIMIT
+#ifndef Py_C_RECURSION_LIMIT
```
Do users define this?

vstinner replied:
I don't know if it's relevant to support that. It's supposed to be a constant built into Python, no?

vstinner committed:
Remove #ifndef Py_C_RECURSION_LIMIT

vstinner committed:
test.pythoninfo logs Py_C_RECURSION_LIMIT

test.pythoninfo logs Py_C_RECURSION_LIMIT and other _testcapi and
_testinternalcapi constants.
